src/validation.cpp

```diff
@@ -4916,3 +4760,21 @@ class CMainCleanup
         mapBlockIndex.clear();
     }
 } instance_of_cmaincleanup;
+
+bool CBlockIndexWorkComparator::operator()(const CBlockIndex *pa, const CBlockIndex *pb) const {
```

Comment:
in commit a8930a66b1 move-only: make the CChainState interface public


Why is this moved to the end of the cpp file instead of keeping it at the top? This breaks both `git blame` and git diff with `--color-moved=dimmed-zebra --ignore-all-space`

Reply:
It'd be in an awkward place at the top of the file - right below the "global state" comment. Happy to move it back if you think the diff convenience is worth it though.

Reply:
I don't understand what the "global state" comment should convey. If a symbol is global and not const, we prefix it with `g_`. Might want to remove it?